core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -1617,7 +1615,7 @@ public void receivePending(Transaction tx, @Nullable List<Transaction> dependenc
             // If this tx spends any of our unspent outputs, mark them as spent now, then add to the pending pool. This
             // ensures that if some other client that has our keys broadcasts a spend we stay in sync. Also updates the
             // timestamp on the transaction and registers/runs event listeners.
-            commitTx(tx);
+            commitTx(tx, false);
```

Comment:
If we die before the autosave, when we start up again, we will start scanning from a block that is at or before where the tx appeared, because the wallet save file contains the last block we processed.  There is no way we can miss a confirmed tx by delaying the save.